src/plugins/__tests__/html.spec.ts

```diff
@@ -0,0 +1,158 @@
+import unified from 'unified';
+import { parse } from '../../parse';
+import mergeHtml from '../mergeHtml';
+
+describe('merge HTML plugin', () => {
+  it('should handle empty element', () => {
+    const parsed = parse(`<span></span>`);
+
+    expect(
+      unified()
+        .use(mergeHtml)
+        .runSync(parsed),
+    ).toStrictEqual(
+      expect.objectContaining({
+        type: 'root',
+        children: [
+          {
+            type: 'paragraph',
+            position: expect.any(Object),
+            children: [
+              {
+                type: 'inlineHtml',
+                attributes: {},
+                tagName: 'span',
+                position: expect.any(Object),
+                children: [],
+              },
+            ],
+          },
+        ],
+      }),
+    );
+  });
+
+  it('should parse attributes', () => {
+    // based on https://github.com/nghiattran/html-attribute-parser/blob/master/test.js#L6
+    const parsed = parse(`<span attr class="animated infinite" ng-class="main.classAnimation"></span>`);
+
+    expect(
+      unified()
+        .use(mergeHtml)
+        .runSync(parsed),
+    ).toStrictEqual(
+      expect.objectContaining({
+        type: 'root',
+        children: [
+          {
+            type: 'paragraph',
+            position: expect.any(Object),
+            children: [
+              {
+                type: 'inlineHtml',
+                attributes: {
+                  attr: true,
+                  class: 'animated infinite',
+                  'ng-class': 'main.classAnimation',
+                },
+                tagName: 'span',
+                position: expect.any(Object),
+                children: [],
+              },
+            ],
+          },
+        ],
+      }),
+    );
+  });
+
+  it('should prepare children for inline html tag', () => {
+    const parsed = parse(`HTML link <a href="https://google.com">**hello** _world_ ~~navigate~~ to</a>.`);
+
+    expect(
+      unified()
+        .use(mergeHtml)
+        .runSync(parsed),
+    ).toStrictEqual(
+      expect.objectContaining({
```

Comment:
Don't these two contradict eachother?

If you want strict checking (which is great), why don't you simply pass an object literal?

This applies to a few other places as well.

Reply:
> Don't these two contradict eachother?

Yup, true, it has no effect when paired with `objectContaining`.

> why don't you simply pass an object literal?

I can do it, I already have all properties listed anyway, so should be fairly easy to do.